src/js/node/http2.ts

```diff
@@ -874,12 +877,13 @@ class ClientHttp2Session extends Http2Session {
     write(self: ClientHttp2Session, buffer: Buffer) {
       if (!self) return;
       const socket = self[bunHTTP2Socket];
-      if (self.#closed) {
-        //queue
-        self.#queue.push(buffer);
-      } else {
+      if (!socket) return;
+      if (self.#connected) {
         // redirect writes to socket
         socket.write(buffer);
+      } else {
+        //queue
+        self.#queue.push(buffer);
       }
```

Comment:
if(self[bunHTTP2Socket])

Reply:
The socket can be attached and not connected yet